src/Controller/QuestionController.php

```diff
@@ -37,4 +44,38 @@ public function index(): string
             'errors' => $errors]
         );
     }
+
+    /**
+     * Displays a message that tells if the user answered the question correctly.
+     * Redirects to index if $_POST['choice'] does not exist or is empty.
+     * @return string
+     */
+    public function result(): string
+    {
+        $message = "";
+        $choiceManager = new ChoiceManager();
+
+        // Check if id is an int and > 0 and displays an error message if id not found in choice table.
+        try {
+            $id = filter_input(INPUT_POST, 'choice', FILTER_VALIDATE_INT, ["options" => ["min_range" => 1]]);
+            if ($id) {
+                $choice = $choiceManager->selectOneById($id);
```

Comment:
You could have a valid id (integer > 0) that is not in the table.
You should check that you get a real choice.